controllers/chat/ChatController.js

```diff
@@ -426,6 +426,135 @@ module.exports = {
+       */
+      const user = {
+        name: `Anonymous ${anon_user_info_emoji_name}`,
+        id: req.userId,
+        image: '',
+        username: `Anonymous ${anon_user_info_emoji_name}`
+      };
+      await client.connectUser(user, req.token);
+
+      if (client.user.name !== `Anonymous ${anon_user_info_emoji_name}`) {
+        await client.upsertUser({id: req.userId, name: `Anonymous ${anon_user_info_emoji_name}`});
+      }
+
+      const channel = client.channel('messaging', {members});
+      const createdChannel = await channel.create();
+
+      await channel.updatePartial({
+        set: {
+          channel_type: CHANNEL_TYPE.ANONYMOUS,
+          anon_user_info_color_code,
+          anon_user_info_color_name,
+          anon_user_info_emoji_code,
+          anon_user_info_emoji_name
+        }
+      });
+
+      // Default mock chat in case of channel is blocked
+      let chat = {
+        message: {
+          id: uuidv4(),
+          cid: createdChannel?.channel?.id,
+          text: message,
+          message,
+          user: {
+            id: req.userId,
+            name: `Anonymous ${anon_user_info_emoji_name}`,
+            image: '',
+            username: `Anonymous ${anon_user_info_emoji_name}`
+          },
+          created_at: new Date(),
+          updated_at: new Date()
+        }
+      };
+
+      if (!createdChannel?.channel?.is_channel_blocked) {
+        chat = await channel.sendMessage({
+          user_id: req.userId,
+          text: message,
+          ...req.body
+        });
+      }
+
+      const targets = members.filter((member) => member !== req.userId);
+      await targets.map(async (target) => {
+        const exist = await ChatAnonUserInfo.count({
+          where: {
+            channel_id: channel.id,
+            my_anon_user_id: req.userId,
+            target_user_id: target
+          }
+        });
+
+        if (!exist) {
+          await ChatAnonUserInfo.create({
+            channel_id: channel.id,
+            my_anon_user_id: req.userId,
+            target_user_id: target,
+            anon_user_info_color_code,
+            anon_user_info_color_name,
+            anon_user_info_emoji_code,
+            anon_user_info_emoji_name
+          });
+        }
+      });
+
+      const targetsUserModel = await UsersFunction.findMultipleUsersById(User, targets);
+      targetsUserModel.push({
+        user_id: req.userId,
+        username: `Anonymous ${anon_user_info_emoji_name}`,
+        profile_pic_path: '',
+        anon_user_info_color_code,
+        anon_user_info_color_name,
+        anon_user_info_emoji_code,
+        anon_user_info_emoji_name
+      });
+
+      // get 100 messages
+      const channelFilters = {cid: 'messaging:' + channel.id};
+      const messageFilters = {created_at: {$lte: new Date()}};
+      const messageHistory = await client.search(channelFilters, messageFilters, {
+        sort: [{updated_at: -1}],
+        limit: 100
+      });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The limit of 100 messages is hardcoded. It would be better to make this a configurable parameter, so you can easily adjust it if needed.

```diff
- limit: 100
+ limit: config.messageHistoryLimit
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->